File: app/src/main/java/com/ed/redditapp/lib/Timer.java
package com.ed.redditapp.lib;

import android.os.Handler;
import android.os.Looper;

public class Timer {
    private final int interval;
    private final Handler handler;
    private final Runnable runnable;

    public Timer(int interval, Runnable runnable) {
        this.interval = interval;
        this.runnable = runnable;
        handler = new Handler(Looper.getMainLooper());
    }

    public void run() {
        handler.removeCallbacks(runnable);
        handler.postDelayed(runnable, interval);
    }
}

File: app/src/main/java/com/ed/redditapp/lib/Timer.kt
package com.ed.redditapp.lib

import android.os.Handler
import android.os.Looper

class Timer(private val interval: Int, private val runnable: Runnable) {
    private val handler = Handler(Looper.getMainLooper())

    fun run() {
        handler.removeCallbacks(runnable)
        handler.postDelayed(runnable, interval.toLong())
    }
}